clearnode/export_transactions_cli.go

```diff
@@ -3,66 +3,140 @@ package main
+	dbTransactions, err := GetLedgerTransactions(e.db, NewAccountID(options.AccountID), options.AssetSymbol, options.TxType)
 	if err != nil {
-		logger.Fatal("Failed to setup database", "error", err)
+		return fmt.Errorf("failed to get transactions: %w", err)
 	}
 
-	transactions, err := GetLedgerTransactions(db, NewAccountID(accountID), "", nil)
-	if err != nil {
-		logger.Fatal("Failed to get transactions", "error", err)
-	}
+	csvWriter := csv.NewWriter(writer)
+	defer csvWriter.Flush()
 
-	if err := os.MkdirAll("csv_export", 0755); err != nil {
-		logger.Fatal("Failed to create directory", "error", err)
-	}
-	fileName := fmt.Sprintf("csv_export/transactions_%s.csv", accountID)
-	file, err := os.Create(fileName)
-	if err != nil {
-		logger.Fatal("Failed to create CSV file", "error", err)
+	// Write header
+	header := []string{"ID", "Type", "FromAccount", "FromAccountTag", "ToAccount", "ToAccountTag", "AssetSymbol", "Amount", "CreatedAt"}
+	if err := csvWriter.Write(header); err != nil {
+		return fmt.Errorf("failed to write header to CSV: %w", err)
 	}
-	defer file.Close()
-
-	writer := csv.NewWriter(file)
-	defer writer.Flush()
 
-	// Write header
-	header := []string{"ID", "Type", "FromAccount", "ToAccount", "AssetSymbol", "Amount", "CreatedAt"}
-	if err := writer.Write(header); err != nil {
-		logger.Fatal("Failed to write header to CSV", "error", err)
+	transactions, err := FormatTransactionsWithTags(e.db, dbTransactions)
+	if err != nil {
+		return fmt.Errorf("failed to format transactions with tags: %w", err)
 	}
 
 	// Write transactions
 	for _, tx := range transactions {
 		row := []string{
-			fmt.Sprintf("%d", tx.ID),
-			tx.Type.String(),
+			fmt.Sprintf("%d", tx.Id),
+			tx.TxType,
 			tx.FromAccount,
+			tx.FromAccountTag,
 			tx.ToAccount,
-			tx.AssetSymbol,
+			tx.ToAccountTag,
+			tx.Asset,
 			tx.Amount.String(),
 			tx.CreatedAt.String(),
 		}
-		if err := writer.Write(row); err != nil {
-			logger.Fatal("Failed to write row to CSV", "error", err)
+		if err := csvWriter.Write(row); err != nil {
+			return fmt.Errorf("failed to write row to CSV: %w", err)
 		}
 	}
+	return nil
+}
+
+// ExportToFile exports transactions to a CSV file
+func (e *TransactionExporter) ExportToFile(options ExportOptions) (string, error) {
+	if err := os.MkdirAll(options.OutputDir, 0755); err != nil {
+		return "", fmt.Errorf("failed to create directory %s: %w", options.OutputDir, err)
+	}
+
+	fileName := filepath.Join(options.OutputDir, fmt.Sprintf("transactions_%s.csv", options.AccountID))
+	file, err := os.Create(fileName)
+	if err != nil {
+		return "", fmt.Errorf("failed to create CSV file %s: %w", fileName, err)
+	}
+	defer file.Close()
 
+	if err := e.ExportToCSV(file, options); err != nil {
+		return "", fmt.Errorf("failed to export to CSV: %w", err)
+	}
+
+	return fileName, nil
+}
+
+func runExportTransactionsCli(logger Logger) {
+	logger = logger.NewSystem("export-transactions")
+	if len(os.Args) < 3 {
```

Comment:
Shouldn't the case `len(os.Args) > 5` return a wrong n-arguments kind of error as well?